fuse_core/include/fuse_core/async_publisher.h

```diff
@@ -135,6 +136,7 @@ class AsyncPublisher : public Publisher
 protected:
   ros::CallbackQueue callback_queue_;  //!< The local callback queue used for all subscriptions
   std::string name_;  //!< The unique name for this publisher instance
+  // rclcpp::Node::SharedPtr node_;  //!< The node for this publisher (TODO(CH3): Uncomment when it's time)
```

Comment:
This might end up being a shared pointer to [the `NodeTopicsInterface`](https://github.com/ros2/rclcpp/blob/rolling/rclcpp/include/rclcpp/node_interfaces/node_topics_interface.hpp) , so that lifecycle nodes can be used too

Reply:
Comment updated https://github.com/locusrobotics/fuse/pull/279/commits/771e4f4b228a1bb1c4dea24ad07fbcd46502ea71

Pending merge of https://github.com/ros2/rclcpp/pull/2041 , but if we expect that to take too long I can duplicate an implementation in fuse_core...

Reply:
Reverting the comment update since it turns out we do have to store a SharedPtr to a node after all (for now at least)